server/base/src/main/java/org/apache/accumulo/server/util/TabletIterator.java

```diff
@@ -101,6 +102,20 @@ public boolean hasNext() {
 
       currentTabletKeys = scanToPrevEndRow();
       if (currentTabletKeys.size() == 0) {
+        if (lastTablet != null) {
+          KeyExtent lastExtent = new KeyExtent(lastTablet, (Text) null);
+
+          if (lastExtent.getEndRow() != null) {
```

Comment:
This part got a little confusing for me. A brief human-friendly comment might be helpful to explain this block's condition:

```suggestion
          // if lastTablet was not the default tablet
          if (lastExtent.getEndRow() != null) {
```